pkg/kube/cluster-init.sh

```diff
@@ -3,143 +3,110 @@
-}
+# Set the node IP to multus differently for single node and cluster mode
+assign_multus_nodeip() {
+  if [ -f /var/lib/edge-node-cluster-mode ]; then
+    NODE_IP=$(get_cluster_node_ip "$1")
+    ClusterPrefixMask=$(get_cluster_prefix_len)
+    ip_prefix=$(ipcalc -n "$NODE_IP$ClusterPrefixMask" | cut -d "=" -f2)
+    ip_prefix="$ip_prefix$ClusterPrefixMask"
+    logmsg "Cluster Node IP prefix to multus: $ip_prefix with node-ip $NODE_IP"
+  else
+    while [ -z "$NODE_IP" ]; do
+      # Find the default route interface
+      default_interface="$(ip route show default | head -n 1 | awk '/default/ {print $5}')"
 
-# kubernetes's name must be lower case and '-' instead of '_'
-convert_to_k8s_compatible() {
-        echo "$1" | tr '[:upper:]_' '[:lower:]-'
-}
+      # Get the first IP address of the default route interface as the node IP
+      NODE_IP="$(ip addr show dev "$default_interface" | awk '/inet / {print $2}' | head -n 1 | cut -d "/" -f1)"
 
-wait_for_device_name() {
-        logmsg "Waiting for DeviceName from controller..."
-        EdgeNodeInfoPath="/persist/status/zedagent/EdgeNodeInfo/global.json"
-        while [ ! -f $EdgeNodeInfoPath ]; do
-                sleep 5
-        done
-        dName=$(jq -r '.DeviceName' $EdgeNodeInfoPath)
-        if [ -n "$dName" ]; then
-                HOSTNAME=$(convert_to_k8s_compatible "$dName")
-        fi
+      [ -z "$NODE_IP" ] && sleep 1
+    done
 
-        # we should have the uuid since we got the device name
-        DEVUUID=$(/bin/hostname)
+    ip_prefix="$NODE_IP/32"
+    logmsg "Single Node IP prefix to multus: $ip_prefix with node-ip $NODE_IP"
+  fi
 
-        if ! grep -q node-name /etc/rancher/k3s/config.yaml; then
-                echo "node-name: $HOSTNAME" >> /etc/rancher/k3s/config.yaml
-        fi
-        logmsg "Hostname: $HOSTNAME"
+  logmsg "Assign node-ip for multus with $ip_prefix"
+  # fill in the outbound external Interface IP prefix in multus config
+  awk -v new_ip="$ip_prefix" '{gsub("IPAddressReplaceMe", new_ip)}1' /etc/multus-daemonset.yaml > /etc/multus-daemonset-new.yaml
 }
 
 apply_multus_cni() {
-        get_default_intf_IP_prefix
-        kubectl create namespace eve-kube-app
-        logmsg "Apply Multus, Node-IP: $NODE_IP"
-        if ! kubectl apply -f /tmp/multus-daemonset.yaml; then
+        # remove get_default_intf_IP_prefix
+        #get_default_intf_IP_prefix
+        if ! kubectl get namespace eve-kube-app > /dev/null 2>&1; then
+                kubectl create namespace eve-kube-app
+        fi
+        logmsg "Apply multus-daemonset-new.yaml"
+        if ! kubectl apply -f /etc/multus-daemonset-new.yaml > /dev/null 2>&1; then
+                logmsg "Apply Multus, has failed, jump out now"
                 return 1
         fi
         logmsg "Done applying Multus"
-        ln -s /var/lib/cni/bin/multus /var/lib/rancher/k3s/data/current/bin/multus
+        link_multus_into_k3s
```

Comment:
`link_multus_into_k3s` is not defined yet, its in an upcoming PR for pkg/kube/cluster-update.sh

Reply:
I can submit a quick PR to add a skeleton cluster-update.sh with just this defined to get this moving again.

Reply:
I'm curious why shellcheck didn't find this

Reply:
I created #4445 to define the missing function